### DISCUSSION TITLE
HTML WEB SAMPLES FOR TOKEN AUTH

### DISCUSSION THREAD

**Discussion by jdfoxito** [fastapi/fastapi#12383] [Questions] [answered]
### Privileged issue

- [X] I'm @tiangolo or he asked me directly to create an issue here.

### Issue Content

Everything demo, o smal sample is oriented to json, not web templates,

could u include samples with html samples, about login jwt many concurrent users.

pls

**Comment by sinisaos** [comment, marked as answer]
@jdfoxito In [this discussion](https://github.com/fastapi/fastapi/discussions/11891#discussioncomment-10195914) you have a repo example of FastAPI authentication with HTML templates. I helped out a bit with a few PRs in that repo that didn't have `CSRF` protection (must have with HTML templates). Now [asgi_csfr](https://github.com/simonw/asgi-csrf) is applied, but I see from this [PR discussion](https://github.com/SWimplifyDev/FastAuthBase/pull/3) that it doesn't work with `FastAPI CLI` (it works directly with `uvicorn`). For everything to work with the `FastAPI CLI`, it is necessary to apply another library, [fastapi-csrf-protect](https://github.com/aekasitt/fastapi-csrf-protect), which I see that the owner of the repo has not applied. I have that application with applied `fastapi-csrf-protect` somewhere on my computer, so if needed I can upload it here as a zip folder (because I don't have it on Github). Basically from that repo you can see how to implement FastaAPI authentication with HTML templates. Hope that helps.
